components/issue-filters/mobile-filters-modal.tsx

```diff
@@ -75,16 +82,24 @@ export default function MobileFiltersModal({
     applyFilters();
   }
 
-  function FilterComponent(label, options, type) {
-    return(
+  function RenderFilterStructure({ label, children }: {label: string, children: ReactElement}) {
```

Comment:
I'd recommend you to move this function to another file and make it a real component, as we have seen that type of component inside a component can lead to re-rendering issues